pkg/customresources/certificatestores/certificatestores.go

```diff
@@ -14,72 +14,82 @@ limitations under the License.
+
 // ActiveCertStores implements the CertStoreManager interface
 type ActiveCertStores struct {
-	// TODO: Implement concurrent safety using sync.Map
-	// The structure of the map is as follows:
-	// The first level maps from scope to certificate stores.
-	// The second level maps from certificate store name to certificates.
+	// scopedCertStores is mapping from cert store name to certificate list.
 	// The certificate store name is prefixed with the namespace.
 	// Example:
 	// {
-	//   "namespace1": {
-	//     "namespace1/store1": []*x509.Certificate,
-	//     "namespace1/store2": []*x509.Certificate
-	//   },
-	//   "namespace2": {
-	//     "namespace2/store1": []*x509.Certificate,
-	//     "namespace2/store2": []*x509.Certificate
-	//   }
+	//   "namespace1/store1": []*x509.Certificate,
+	//   "namespace2/store2": []*x509.Certificate
 	// }
-	// Note: Scope is utilized for organizing and isolating cert stores. In a Kubernetes (K8s) environment, the scope can be either a namespace or an empty string ("") for cluster-wide cert stores.
-	ScopedCertStores map[string]map[string][]*x509.Certificate
+	// Note: The namespace "default" is reserved for cluster-wide scenario.
+	scopedCertStores sync.Map
 }
 
 func NewActiveCertStores() CertStoreManager {
-	return &ActiveCertStores{
-		ScopedCertStores: make(map[string]map[string][]*x509.Certificate),
-	}
+	return &ActiveCertStores{}
 }
 
 // GetCertStores fulfills the CertStoreManager interface.
-// It returns a list of cert stores for the given scope. If no cert stores are found for the given scope, it returns cluster-wide cert stores.
-// TODO: Current implementation always fetches cluster-wide cert stores. Will support actual namespaced certStores in future.
-func (c *ActiveCertStores) GetCertStores(_ string) map[string][]*x509.Certificate {
-	return c.ScopedCertStores[constants.EmptyNamespace]
+// It returns a list of certificates in the given store.
+func (c *ActiveCertStores) GetCertsFromStore(ctx context.Context, storeName string) []*x509.Certificate {
+	storeName = prependNamespaceToStoreName(storeName)
+	if !isCompatibleNamespace(ctx, storeName) {
+		return []*x509.Certificate{}
+	}
+	if certs, ok := c.scopedCertStores.Load(storeName); ok {
+		return certs.([]*x509.Certificate)
+	}
+	return []*x509.Certificate{}
 }
 
 // AddStore fulfills the CertStoreManager interface.
-// It adds the given certificate under the given scope.
-// TODO: Current implementation always adds the given certificate to cluster-wide cert store. Will support actual namespaced certStores in future.
-func (c *ActiveCertStores) AddStore(_, storeName string, certs []*x509.Certificate) {
-	scope := constants.EmptyNamespace
-	if c.ScopedCertStores[scope] == nil {
-		c.ScopedCertStores[scope] = make(map[string][]*x509.Certificate)
-	}
-	c.ScopedCertStores[scope][storeName] = certs
+// It adds the given certificate under cert store.
+func (c *ActiveCertStores) AddStore(storeName string, cert []*x509.Certificate) {
```

Comment:
general comment: I want to double check some questions:
- THe certificate store will remain namespaced right?
- If a user provides a namespaced reference in the cluster verifier to a certificate store, it will be matched to a namespaced certificate store since the namespace is still honored even though it's a clustered `Verifier` resource?
- What happens in this scenario? a user prior to 1.2 had a non-namespaced reference to a certificate store in the ratify install namespace. After upgrade, the user is still using Verifier resource but now it's cluster scoped. They expect that previous non-namespaced reference will still point to the ratify installed namespace certificate store resource. Is that the case?

Reply:
Thanks for double checking!
1. Right, certStore is always namespaced without change.
2. Right, the certStore could be either gatekeeper-system/certStore if it's not prefixed with namespace or namespace/certStore if users provide the namespace.
3. Correct. It will try to fetch gatekeeper-system/certs as we discussed in another comment.

I understand your confusion which is one key point in this PR. I'll remove prepending `default` namespace logic so that the behavior keeps the same as v1.1.

Reply:
Thanks for addressing the concerns